### PR TITLE
CQ-4342644 Core Components Runtime

### PR DESCRIPTION
**Pull request by nit23uec**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

File: bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/form/FormConstants.java [hunk 1]
```diff
@@ -34,6 +34,9 @@ private FormConstants() {
     /** The resource type for text input v1 */
     public static final String RT_FD_FORM_TEXT_V1 = RT_FD_FORM_PREFIX + "textinput/v1/textinput";
 
+    /** The resource type for date picker v1 */
+    public static final String RT_FD_FORM_DATE_PICKER_V1 = RT_FD_FORM_PREFIX + "datepicker/v1/datepicker";
```
Comment (rismehta) [inline bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/form/FormConstants.java:38]: Can we remove this constant and use it directly where it is required ? With constants for resource type, this file would grow exponentially

Comment (nit23uec) [reply bundles/core/src/main/java/com/adobe/cq/forms/core/components/internal/form/FormConstants.java:38]: not exponentially. just 20 lines. I'd suggest to keep it as it is.

**Review by rismehta — CHANGES_REQUESTED** [review]
I am still reviewing, I would add more comments shortly

File: ui.frontend/package.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    "aem-clientlib-generator": "^1.8.0",
+    "html-webpack-plugin": "^5.5.0",
+    "jest": "^28.1.1",
+    "jest-environment-jsdom": "^28.1.1",
+    "webpack": "^5.73.0",
+    "webpack-cli": "^4.10.0"
+  },
+  "dependencies": {
+    "@aemforms/af-core": "^0.15.0",
+    "jquery": "^3.6.0"
```
[the remark this answers is not in the document]
Comment (nit23uec) [reply ui.frontend/package.json:24]: this is to show that its fine to use jquery in the component runtime for things like things like setValue.
` $(this.element).find("input[type=date]").val(value);`

File: ui.frontend/__tests__/datepicker.test.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+ * limitations under the License.
+ ******************************************************************************/
+import * as datePickerFormJson from "./mocks/formModel.json";
+import FormContainer from "../src/view/FormContainer";
+import Form from "@aemforms/af-core/lib/Form";
+import DatePicker from "../src/view/DatePicker";
+
+test('datepicker test', () => {
+    document.body.innerHTML = `
+    <div class="cmp-datepicker" data-cmp-is="datepicker" data-cmp-formcontainer="/content/forms/af/try1/jcr:content/guideContainer">
```
[the remark this answers is not in the document]
Comment (nit23uec) [reply ui.frontend/__tests__/datepicker.test.js:23]: ETE test will also be there.
this is just for unit testing with minimal markup.

**Review by rismehta — CHANGES_REQUESTED** [review]
Let's not merge this for now. Shared few thoughts with @vdua

**Pull request closed without merging**